codice/acdebugger
# ACD-22 Adding test cases

### Description of the Change
Bug fixed and code modification to make it more easily testable are provided in first commit.
Spock test cases and code coverage updates provided in second commit.

### Benefits
The whole AC debugger is now fully unit tested.

### Possible Drawbacks
Lots of test cases to review 😉 

### Verification Process
- run the full build
- verify manually the debugger in a running context

### Applicable Issues
Fixes: #22
